### PR TITLE
Code cleanup

### PR DESCRIPTION
**Pull request by neodesys**
- Simplify Java classes removing uneeded glue
- Improve native code (fix warnings, remove obsolete code and organize includes)
- Use new JNI wrapper in native code
- Activate clang-tidy for the whole code

### CODE REVIEW

File: wpe/src/main/cpp/Browser/Browser.h [hunk 1]
```diff
@@ -22,50 +22,41 @@
...
     Browser(Browser&&) = delete;
     Browser& operator=(Browser&&) = delete;
     Browser(const Browser&) = delete;
     Browser& operator=(const Browser&) = delete;
 
-    ~Browser() { shut(); }
+    ~Browser() { jniShut(); }
 
-    WebKitWebContext* webContext() const { return m_webContext; }
+    WebKitWebContext* getWebContext() const noexcept { return m_webContext.get(); }
```
Comment (zhani) [inline wpe/src/main/cpp/Browser/Browser.h:46]: this is not according to webkit code style

https://webkit.org/code-style-guidelines/#names-setter-getter

File: wpe/src/main/cpp/Browser/ExportedBuffer.h [hunk 2]
```diff
@@ -57,7 +44,22 @@ struct ExportedBuffer final : public std::enable_shared_from_this<ExportedBuffer
...
     ~ExportedBuffer()
     {
-        if (buffer)
-            AHardwareBuffer_release(buffer);
+        if (m_buffer != nullptr)
+            AHardwareBuffer_release(m_buffer);
     }
+
+    AHardwareBuffer* getBuffer() const noexcept { return m_buffer; }
+    uint32_t getPoolId() const noexcept { return m_poolId; }
```
Comment (zhani) [inline wpe/src/main/cpp/Browser/ExportedBuffer.h:52]: not according to style guide

https://webkit.org/code-style-guidelines/#names-setter-getter

File: wpe/src/main/cpp/Browser/ExportedBuffer.h [hunk 3]
```diff
@@ -57,7 +44,22 @@ struct ExportedBuffer final : public std::enable_shared_from_this<ExportedBuffer
...
+    uint32_t getBufferId() const noexcept { return m_bufferId; }
```
Comment (zhani) [inline wpe/src/main/cpp/Browser/ExportedBuffer.h:53]: ditto

File: wpe/src/main/cpp/Browser/ExportedBuffer.h [hunk 4]
```diff
@@ -57,7 +44,22 @@ struct ExportedBuffer final : public std::enable_shared_from_this<ExportedBuffer
...
+    uint32_t getWidth() const noexcept { return m_size.m_width; }
```
Comment (zhani) [inline wpe/src/main/cpp/Browser/ExportedBuffer.h:54]: ditto

File: wpe/src/main/cpp/Browser/ExportedBuffer.h [hunk 5]
```diff
@@ -57,7 +44,22 @@ struct ExportedBuffer final : public std::enable_shared_from_this<ExportedBuffer
...
+    uint32_t getHeight() const noexcept { return m_size.m_height; }
```
Comment (zhani) [inline wpe/src/main/cpp/Browser/ExportedBuffer.h:55]: ditto

File: wpe/src/main/cpp/Browser/LooperThread.h [hunk 6]
```diff
@@ -20,17 +20,28 @@
...
+    LooperThread& operator=(LooperThread&&) = delete;
+    LooperThread(const LooperThread&) = delete;
+    LooperThread& operator=(const LooperThread&) = delete;
 
     ~LooperThread();
 
-    ALooper* looper() const { return m_looper; }
+    void startLooper() noexcept;
+
+    ALooper* getLooper() const noexcept { return m_looper; }
```
Comment (zhani) [inline wpe/src/main/cpp/Browser/LooperThread.h:42]: wrong style

https://webkit.org/code-style-guidelines/#names-setter-getter

File: wpe/src/main/cpp/Browser/Page.h [hunk 7]
```diff
@@ -23,76 +23,45 @@
...
-    void deleteInputMethodContent(int offset);
+    ~Page() override { close(); }
 
-    void domFullscreenRequest(bool fullscreen);
-    void requestExitFullscreen();
-    void fullscreenImageReady();
+    void close() noexcept;
 
-    struct wpe_android_view_backend_exportable* exportable() { return m_viewBackendExportable; }
+    WebKitWebView* getWebView() const noexcept { return m_webView; }
```
Comment (zhani) [inline wpe/src/main/cpp/Browser/Page.h:47]: "get" in getter name is wrong according to webkit style guide

**Review by zhani — CHANGES_REQUESTED** [review]

File: wpe/src/main/cpp/Browser/Browser.h (hunk 1 above)
Comment (neodesys) [reply wpe/src/main/cpp/Browser/Browser.h:46]: I was trying to be coherent with the Android/Java part, but you're right.
I'm fixing it.

File: wpe/src/main/cpp/Browser/ExportedBuffer.h (hunk 2 above)
Comment (neodesys) [reply wpe/src/main/cpp/Browser/ExportedBuffer.h:52]: done

File: wpe/src/main/cpp/Browser/ExportedBuffer.h (hunk 3 above)
Comment (neodesys) [reply wpe/src/main/cpp/Browser/ExportedBuffer.h:53]: done

File: wpe/src/main/cpp/Browser/LooperThread.h (hunk 6 above)
Comment (neodesys) [reply wpe/src/main/cpp/Browser/LooperThread.h:42]: done

File: wpe/src/main/cpp/Browser/Page.h (hunk 7 above)
Comment (neodesys) [reply wpe/src/main/cpp/Browser/Page.h:47]: done

**Review by zhani — APPROVED** [review]

**Pull request merged**